src/app/pages/send-tokens/components/send-form-inner.tsx

```diff
@@ -17,27 +20,42 @@ import { useTransferableAssets } from '@app/store/assets/asset.hooks';
 import { useSelectedAsset } from '@app/pages/send-tokens/hooks/use-selected-asset';
 import { RecipientField } from '@app/pages/send-tokens/components/recipient-field';
 import { MemoField } from '@app/pages/send-tokens/components/memo-field';
-import { useSendFormUnsignedTxPreviewState } from '@app/store/transactions/transaction.hooks';
+import {
+  useSendFormEstimatedUnsignedTxByteLengthState,
+  useSendFormSerializedUnsignedTxPayloadState,
+  useSendFormUnsignedTxPreviewState,
+} from '@app/store/transactions/transaction.hooks';
 import { LoadingRectangle } from '@app/components/loading-rectangle';
-import { FeeEstimate } from '@shared/models/fees-types';
+import { useFeeEstimations } from '@app/query/fees/fees.hooks';
+import { FeeCalculationType } from '@shared/models/fees-types';
 import { SendFormSelectors } from '@tests/page-objects/send-form.selectors';
 
 import { SendFormMemoWarning } from './memo-warning';
 
 interface SendFormInnerProps {
   assetError: string | undefined;
-  feeEstimations: FeeEstimate[];
   nonce: number | undefined;
+  setAssetIdForValidation(assetId: string): void;
+  setFeeCalculationType(type: FeeCalculationType): void;
```

Comment:
As these are events firing upwards, i'd name them as event actions like `onAssetIdSelected`. The fact that it's for validation, is no concern of `<SendFormInner />`.

For `setFeeCalculationType`, upon which event do we want to set fee calculation type?

Reply:
Ok, I can rename, makes sense. `setFeeCalculationType` is used when we submit the tx and record the event. It just needs to be set once the fee estimations are loaded.

Reply:
In this case would rename `onFeeEstimationsLoaded` or something similar